Lib/test/libregrtest/runtest.py

```diff
@@ -440,14 +440,9 @@ def _load_run_test(result: TestResult, ns: Namespace) -> None:
 
     test_mod = importlib.import_module(module_name)
 
-    # If the test has a test_main, that will run the appropriate
-    # tests.  If not, use normal unittest test runner.
-    test_main = getattr(test_mod, "test_main", None)
-    if test_main is not None:
-        test_func = test_main
-    else:
-        def test_func():
-            return run_unittest(test_mod)
+    assert not hasattr(test_mod, "test_main")
```

Comment:
Can you please replace the assert with a regular exception. Like:

```py
if hasattr(test_mod, "test_main"):
    raise Exception("Module {result.test_name} defines test_main() which is no longer supported by regrtest")
```

Maybe add a comment linking to ``https://github.com/python/cpython/issues/89392``.